docbrown/core/src/tgraph.rs

```diff
@@ -301,54 +294,169 @@ impl TemporalGraph {
+        let w = w.clone();
+        let mut vs = self.index.range(w.clone()).flat_map(|(_, vs)| vs.iter());
+
+        // First check if v1 exists within the given window
+        match vs.contains(&v1_pid) {
+            true => match &self.adj_lists[*v1_pid] {
+                Adj::Solo(_) => Option::<EdgeView>::None,
+                Adj::List {
+                    out,
+                    remote_out,
+                    into,
+                    remote_into,
+                    ..
+                } => {
+                    // Then check if v2 exists in the given window while sharing an edge with v1
+                    if !self.has_vertex(v2) {
+                        match (
+                            remote_out.find_window(v2 as usize, &w),
+                            remote_into.find_window(v2 as usize, &w),
+                        ) {
+                            (Some(e), None) => Some(EdgeView {
+                                src_g_id: v1,
+                                dst_g_id: v2,
+                                src_id: *v1_pid,
+                                dst_id: v2 as usize,
+                                t: None,
+                                e_meta: e,
+                            }),
+                            (None, Some(e)) => Some(EdgeView {
+                                src_g_id: v2,
+                                dst_g_id: v1,
+                                src_id: v2 as usize,
+                                dst_id: *v1_pid,
+                                t: None,
+                                e_meta: e,
+                            }),
+                            _ => None,
+                        }
+                    } else {
+                        let v2_pid = self.logical_to_physical.get(&v2)?;
+                        match (out.find_window(*v2_pid, &w), into.find_window(*v2_pid, &w)) {
+                            (Some(e), None) => Some(EdgeView {
+                                src_g_id: v1,
+                                dst_g_id: v2,
+                                src_id: *v1_pid,
+                                dst_id: *v2_pid,
+                                t: None,
+                                e_meta: e,
+                            }),
+                            (None, Some(e)) => Some(EdgeView {
+                                src_g_id: v2,
+                                dst_g_id: v1,
+                                src_id: *v2_pid,
+                                dst_id: *v1_pid,
+                                t: None,
+                                e_meta: e,
+                            }),
+                            _ => None,
+                        }
+                    }
+                }
+            },
+            false => Option::<EdgeView>::None,
+        }
+    }
+
+    // FIXME: all the functions using global ID need to be changed to use the physical ID instead
+    pub(crate) fn edges(
```

Comment:
This probably needs to be called something like vertex_edges to make it a bit clearer its for a singular node.